src/OpenTelemetry.Exporter.Geneva/Metrics/Transport/MetricUnixUserEventsDataTransport.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#if NET6_0_OR_GREATER
+
+#nullable enable
+
+using System.Text;
+using Microsoft.LinuxTracepoints.Provider;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal sealed class MetricUnixUserEventsDataTransport : IMetricDataTransport
+{
+    public const uint MetricsProtocol = 0U;
+    public const string MetricsVersion = "v0.19.00";
+    public const string MetricsTracepointName = "otlp_metrics";
+    public const string MetricsTracepointNameArgs = $"{MetricsTracepointName} u32 protocol;char[8] version;__rel_loc u8[] buffer";
+
+    private static readonly ReadOnlyMemory<byte> MetricsVersionUtf8 = Encoding.UTF8.GetBytes(MetricsVersion);
+    private readonly PerfTracepoint metricsTracepoint;
+
+    private MetricUnixUserEventsDataTransport()
+    {
+        this.metricsTracepoint = new PerfTracepoint(MetricsTracepointNameArgs);
+        if (this.metricsTracepoint.RegisterResult != 0)
+        {
+            ExporterEventSource.Log.TransportError(
+                nameof(MetricUnixUserEventsDataTransport),
+                $"Tracepoint for 'otlp_metrics' user events could not be registered: '{this.metricsTracepoint.RegisterResult}'");
+        }
+    }
+
+    public static MetricUnixUserEventsDataTransport Instance { get; } = new();
+
+    public void Send(MetricEventType eventType, byte[] body, int size)
+    {
+        throw new NotSupportedException();
+    }
+
+    public void SendOtlpProtobufEvent(byte[] body, int size)
+    {
+        if (this.metricsTracepoint.IsEnabled)
+        {
+            var buffer = new ReadOnlySpan<byte>(body, 0, size);
+
+            var bufferRelLoc = 0u | ((uint)buffer.Length << 16);
+
+            this.metricsTracepoint.Write(
+                [MetricsProtocol],
+                MetricsVersionUtf8.Span,
+                [bufferRelLoc],
+                buffer);
+        }
```

Comment:
can we do a log on the else part? Given user-events is still very new, if the listener is not configured correctly, metrics won't flow. So some log here is good to quickly indiate that exporter is not bothering to send it due to lack of interested listeners.

Reply:
Mind if I come back to this one? There are some refactors/changes I would like to make but didn't on this PR to keep the diff small.